diff --git src/rt/hoedown/src/document.c.old src/rt/hoedown/src/document.c
index a9cb42c..14a1273 100644
--- src/rt/hoedown/src/document.c.old
+++ src/rt/hoedown/src/document.c
@@ -1191,7 +1191,8 @@ char_link(hoedown_buffer *ob, hoedown_document *doc, uint8_t *data, size_t offse
 			}
 			else if (data[i] == ')') {
 				if (nb_p == 0) break;
-				else nb_p--; i++;
+				else nb_p--;
+				i++;
 			} else if (i >= 1 && _isspace(data[i-1]) && (data[i] == '\'' || data[i] == '"')) break;
 			else i++;
 		}
